test_files/unit_tests/test_symmetric_site.py

```
@@ -0,0 +1,250 @@
+def test_iter_yields_constituent_sites(symmetric_site):
+    """Verify iteration exposes constituent Sites in stored order for uniform processing."""
+    assert tuple(symmetric_site) == symmetric_site.sites
+    assert all(isinstance(site, Site) for site in symmetric_site)
+
+
+def test_symmetric_sites_collection_is_immutable(symmetric_site):
+    """Constituent Sites cannot be added, removed, or reordered."""
+    assert isinstance(symmetric_site.sites, tuple)
+
+    with pytest.raises(AttributeError):
+        symmetric_site.sites.append(site)
```

Ruff (0.16.1)

[error] Undefined name `site`

(F821)